kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/splash/presenter/SplashPresenter.java

```diff
@@ -31,65 +75,113 @@ public void onAttach(ISplashView view) {
-        animateLoading();
-        activateAccount();
-    }
-
-    private void animateLoading() {
-        if (view != null) {
-            view.animateLoading();
-        }
-    }
-
-    private void activateAccount() {
-        authRepository.activateAccount(new KinCallback<Void>() {
-            @Override
-            public void onResponse(Void response) {
-                confirmedSucceed = true;
-                navigateToMarketplace();
-            }
-
-            @Override
-            public void onFailure(KinEcosystemException exception) {
-                showToast("Oops something went wrong...");
-                stopLoading(true);
-            }
-        });
-    }
-
-    private void stopLoading(boolean reset) {
-        if (view != null) {
-            view.stopLoading(reset);
-        }
-    }
-
-    private void showToast(String msg) {
-        if (view != null) {
-            view.showToast(msg);
-        }
-    }
-
-    private void navigateToMarketplace() {
-        if (animationEnded && confirmedSucceed) {
-            if (view != null) {
-                view.navigateToMarketPlace();
-            }
-        }
-    }
-
-    @Override
-    public void backButtonPressed() {
-        eventLogger.send(BackButtonOnWelcomeScreenPageTapped.create());
-        if (view != null) {
-            view.navigateBack();
-        }
-    }
-
-    @Override
-    public void onAnimationEnded() {
-        animationEnded = true;
-        navigateToMarketplace();
-    }
+	public void onDetach() {
+		super.onDetach();
+		removeAccountStateObserver();
+		cancelTimeoutTask();
+	}
+
+	private void removeAccountStateObserver() {
+		new Log().withTag(TAG).text("removeAccountStateObserver").log();
+		accountManager.removeAccountStateObserver(accountStateObserver);
+	}
+
+	@Override
+	public void getStartedClicked() {
+		eventLogger.send(WelcomeScreenButtonTapped.create());
+		isAccountActivated = authRepository.isActivated();
+		animateLoading();
+		new Log().withTag(TAG).text("getStartedClicked")
+			.put("isActivate", isAccountCreated)
+			.put("accountState", accountManager.getAccountState())
+			.log();
+
+		if (accountManager.getAccountState() != CREATION_COMPLETED) {
+			new Log().withTag(TAG).text("addAccountStateObserver").log();
+			accountManager.addAccountStateObserver(accountStateObserver);
+			startCreationTimeout(TIME_OUT_DURATION);
+		} else {
+			isAccountCreated = true;
+		}
+
+		if (!isAccountActivated) {
+			activateAccount();
+		}
+	}
+
+	private void cancelTimeoutTask() {
+		timeOutTask.cancel();
+		timer.purge();
+	}
+
+	private void startCreationTimeout(final int sec) {
+		timer.schedule(timeOutTask, sec * SEC_IN_MILI);
+	}
+
+	private void showTryAgainLater() {
+		showToast(TRY_AGAIN);
+	}
+
+	private void showSomethingWentWrong() {
+		showToast(SOMETHING_WENT_WRONG);
+	}
+
+	private void animateLoading() {
+		if (view != null) {
+			view.animateLoading();
+		}
+	}
+
+	private void activateAccount() {
```

Comment:
the account activation should also be part of the AccountManger (we might want to activate account from different presenter in the future)